dpnp/backend/extensions/statistics/common.hpp

```diff
@@ -101,26 +114,25 @@ struct IsNan
 {
     static bool isnan(const T &v)
     {
-        if constexpr (std::is_floating_point_v<T> ||
-                      std::is_same_v<T, sycl::half>) {
-            return sycl::isnan(v);
+        if constexpr (type_utils::is_complex_v<T>) {
+            const auto real1 = std::real(v);
+            const auto imag1 = std::imag(v);
+
+            using vT = typename T::value_type;
+
+            return IsNan<vT>::isnan(real1) || IsNan<vT>::isnan(imag1);
+        }
+        else {
+            if constexpr (std::is_floating_point_v<T> ||
+                          std::is_same_v<T, sycl::half>) {
+                return sycl::isnan(v);
+            }
         }
 
         return false;
```

Comment:
I believe it would be more proper:
```suggestion
        else if constexpr (std::is_floating_point_v<T> ||
                          std::is_same_v<T, sycl::half>) {
            return sycl::isnan(v);
        }
        else {
            return false;
        }
```